### PR TITLE
Implemented: predefined and custom option for selecting job run time (#85zrxc0xf)

### LINKED ISSUE #430: Job Runtime should be selectable from predefined options
## What is the motivation for adding/enhancing this feature?

Right now, to set runtime, you have to select the date and time from an elaborate picker. sometimes you just want a job to be scheduled without thinking about the time. An option for specific run time should be available as a "custom option" but the first options should be some defaults.

Screenshot 2023-04-07 at 11 17 08 AM

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #430 

### Screenshots of Visual Changes before/after (If There Are Any)

[Screencast from 24-04-23 03:35:32 PM IST.webm](https://user-images.githubusercontent.com/59442907/233966089-a807fc6c-e927-4e86-a43d-b3e1da3a5ea1.webm)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

**Comment by k2maan** [comment]
If you add a custom date time but then change to a predefined time, the custom time is not removed from the select options list.

File: src/components/JobConfiguration.vue [hunk 1]
```diff
@@ -17,7 +17,9 @@
       <ion-item>
         <ion-icon slot="start" :icon="timeOutline" />
         <ion-label class="ion-text-wrap">{{ $t("Run time") }}</ion-label>
-        <ion-label class="ion-text-wrap" @click="() => isOpen = true" slot="end">{{ runTime ? getTime(runTime) : $t('Select run time') }}</ion-label>
+        <ion-select interface="popover" placeholder="Select" :value="runTime.value" @ionChange="updateRunTime($event)">
```
Comment (ymaheshwari1) [inline src/components/JobConfiguration.vue:20]: Internationalize text.

File: src/components/JobConfiguration.vue [hunk 2]
```diff
@@ -17,7 +17,9 @@
...
+          <ion-select-option v-for="runTime in runTimeOptions" :key="runTime.value" :value="runTime.value">{{ $t(runTime.label) }}</ion-select-option>
```
Comment (ymaheshwari1) [inline src/components/JobConfiguration.vue:21]: Add the hardcoded options in en.json

File: src/components/JobConfiguration.vue [hunk 3]
```diff
@@ -329,13 +356,24 @@ export default defineComponent({
...
-      const setTime = handleDateTimeInput(ev['detail'].value);
-      if(setTime > currTime) {
-        this.runTime = setTime;
+    updateRunTime(event: CustomEvent) {
+      const value = event.detail.value
+      if (value != 'CUSTOM') {
+        // check for the first 5, predefined options
+        const isPrefinedValue = this.runTimeOptions.slice(0, 5).some((option: any) => option.value === value)
+
+        if (this.runTime.type == 'CUSTOM'
```
Comment (ymaheshwari1) [inline src/components/JobConfiguration.vue:365]:
```suggestion
        if (this.runTime.type === 'CUSTOM'
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobConfiguration.vue [hunk 4]
```diff
@@ -391,6 +431,25 @@ export default defineComponent({
...
+        this.runTimeOptions.push({
+          value: setTime,
+          label: this.getTime(setTime)
+        })
+
+        this.runTime = {
+          value: setTime,
+          type: 'CUSTOM'
+        }
+      } else {
```
Comment (adityasharma7) [inline src/components/JobConfiguration.vue:450]: I think custom option will remain selected in this case

**Pull request merged**